### PR TITLE
Setup of basic collection rehydration cluster similar to Clive is completed.

### PR DESCRIPTION
**Pull request by umairgha**
Setup of basic collection rehydration cluster similar to Clive is completed.

### CODE REVIEW

File: cluster_config.tf [hunk 1]
```diff
@@ -0,0 +1,106 @@
+resource "aws_emr_security_configuration" "ebs_emrfs_em" {
+  name          = "dataworks_aws_collections_rehydration_ebs_emrfs"
+  configuration = jsonencode(local.ebs_emrfs_em)
+}
+
+#TODO remove this
+output "security_configuration" {
```
Comment (tobywalls) [inline cluster_config.tf]: Is this your "Todo"?

File: data.tf [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    actions = [
+      "s3:Get*",
+      "s3:List*",
+      "s3:DeleteObject*",
+      "s3:Put*",
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.common.outputs.published_bucket.arn}/pdm-dataset/*",
```
Comment (tobywalls) [inline data.tf:24]: Do you need delete on pdm, analytical dataset, entire data/ subdir?

File: processed_bucket.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    actions = [
+      "s3:GetObject*",
+      "s3:DeleteObject*",
+      "s3:PutObject*",
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.common.outputs.processed_bucket.arn}/*"
+    ]
```
Comment (tobywalls) [inline processed_bucket.tf:24]: Seems excessive

File: cluster_config.tf (hunk 1 above)
Comment (umairgha) [reply cluster_config.tf]: I think I shell remove it if it is not breaking anything

File: variables.tf [hunk 4]
```diff
@@ -1,10 +1,50 @@
-variable "assume_role" {
-  type        = string
-  default     = "ci"
-  description = "IAM role assumed by Concourse when running Terraform"
```
Comment (tobywalls) [inline variables.tf:4]: Is this not required for the pipeline?

File: processed_bucket.tf (hunk 3 above)
Comment (tobywalls) [reply processed_bucket.tf:24]: Does this duplicate functionality in data.tf?

File: data.tf (hunk 2 above)
Comment (umairgha) [reply data.tf:24]: Good find, I planned to review it in next iteration, but as its sensitive I will update this shortly

File: processed_bucket.tf (hunk 3 above)
Comment (umairgha) [reply processed_bucket.tf:24]: Yes, the write access would be removed.

File: variables.tf (hunk 4 above)
Comment (umairgha) [reply variables.tf:4]: Its defined in Jinja template

File: data.tf (hunk 2 above)
Comment (umairgha) [reply data.tf:24]: Delete and write access removed from all other location but dataworks-aws-collections-rehydration

File: processed_bucket.tf (hunk 3 above)
Comment (umairgha) [reply processed_bucket.tf:24]: Write access removed

File: cluster_config.tf (hunk 1 above)
Comment (umairgha) [reply cluster_config.tf]: removed

**Review by tobywalls — APPROVED** [review]

**Pull request merged**